program/src/processor.rs

```diff
@@ -543,27 +582,26 @@ fn process_process_proposal(program_id: &Pubkey, accounts: &[AccountInfo]) -> Pr
     let governance_info = next_account_info(accounts_iter)?;
 
     check_governance_exists(program_id, governance_info)?;
-    check_proposal_exists(program_id, proposal_info)?;
 
-    {
-        // Ensure the proposal meets the acceptance threshold.
-        let proposal_data = proposal_info.try_borrow_data()?;
-        let proposal_state = bytemuck::try_from_bytes::<Proposal>(&proposal_data)
-            .map_err(|_| ProgramError::InvalidAccountData)?;
+    let governance_data = governance_info.try_borrow_data()?;
+    let governance_config = bytemuck::try_from_bytes::<Config>(&governance_data)
+        .map_err(|_| ProgramError::InvalidAccountData)?;
 
-        let governance_data = governance_info.try_borrow_data()?;
-        let governance_config = bytemuck::try_from_bytes::<Config>(&governance_data)
-            .map_err(|_| ProgramError::InvalidAccountData)?;
+    check_proposal_exists(program_id, proposal_info)?;
 
-        let clock = <Clock as Sysvar>::get()?;
+    let mut proposal_data = proposal_info.try_borrow_mut_data()?;
+    let proposal_state = bytemuck::try_from_bytes_mut::<Proposal>(&mut proposal_data)
+        .map_err(|_| ProgramError::InvalidAccountData)?;
 
-        proposal_state.check_cooldown(governance_config.cooldown_period_seconds, &clock)?;
+    let clock = <Clock as Sysvar>::get()?;
 
-        // Process the proposal instruction.
-        // TODO!
-    }
+    proposal_state.check_cooldown(governance_config.cooldown_period_seconds, &clock)?;
 
-    close_proposal_account(proposal_info)?;
+    // Process the proposal instruction.
+    // TODO!
+
+    // Set the proposal's status to processed.
+    proposal_state.status = ProposalStatus::Processed;
```

Comment:
It might not be strictly needed, but to be safe, this should check that the proposal is in the `Accepted` / `Cooldown` state

Reply:
In that case, let's just remove it in the meantime, that way it'll make it easier to review the PR which does add it